Add goals to create general stats and synonym stats and rare and community stats

twhetzel opened the pull request:
closes #8809
closes #8906 
closes #8916 
closes #8908

Create general stats as requested in this [gSheet](https://docs.google.com/spreadsheets/d/1TGD1ta2RyoLMvAne1dp-b3LTj6b5Y_jTvsz04HPHmEk/edit?gid=1264043076#gid=1264043076)
The queries can either be run individually or as: 
`sh run.sh make create-general-mondo-stats-all`
- [x] curator review and approval of general statistics values
- [ ] developer review of general statistics queries and make goals

---

Update: added make goal and query to create synonym stats. The query can either be run as a "report" or as
`sh run.sh make create-synonym-mondo-stats-all`
- [x] curator review and approval of synonym statistics values
- [ ] developer review of  synonym statistics queries and make goals

---

Update 27Mar2025: added make goal and query to create rare subset stats. The query can either be run as a "report" or as `sh run.sh make create-rare-mondo-stats-all`
- [x] curator review and approval of rare statistics values
- [ ] developer review of rare statistics queries and make goals

---
Update 29Mar2025: added make goal to create report file of opened and closed issues between two dates with the count within each open/closed issue by tag. The dates can be entered manually or by default it uses the last two most recent dates of the tagged releases compared to the date the goal is run on. It also creates a file of unique users, the count of issues they opened or closed, and unique tags for these issues is also created. 
This can be run from the command line given some prerequisites since it does not use ODK or from the GitHub Action "Generate GitHub Issue Statistics"
- [ ] curator review
Example:
```
# Stats generated on 2025-03-30 06:02:16Z for GitHub issues from 2025-03-15 to 2025-03-18
type	label	count
new_issue		1
closed_issue		12
label_new	New term request	1
label_new	user request	1
label_closed	New term request	2
...
```

and 

```
# Stats generated on 2025-03-30 04:58:22Z for GitHub issues from 2025-03-15 to 2025-03-18
type	github_handle	count	labels
opened	J-Siew	1	New term request,user request
closed	galyea123	3	effort-XS,relabel term,synonym,user request
closed	sagehrke	1	outreach
closed	eedoh01	1	New term request,user request
...
```

twhetzel commented on src/ontology/mondo.Makefile:
```diff
@@ -265,6 +265,56 @@ create-mondo-stats:
+	$(SPARQLDIR)/reports/COUNT-human-genetic-diseases.sparql \
+	$(SPARQLDIR)/reports/COUNT-non-human-genetic-diseases.sparql
+
+TMP_RESULTS_DIR = $(MONDO_STATS_REPORTS_DIR)/tmp
```
yes, it is defined at the top of the `mondo.Makefile`

sabrinatoro reviewed (changes requested):
I ran these queries, and found the following: 

1) total count doesn't match: 

Queries | count
-- | --
AllDiseaseExcludingSusceptibility | 25379
HumanDisease | 25661
NonHumanAnimalDisease | 2959

Human disease + non-human disease are not equal to all disease excluding susceptibility. 
Is it possible that "susceptibilities" are counted in the "human disease" count? if yes, then let's update to human disease = human disease excluding susceptibilities

2) 
Queries  |  counts
-- | --
HumanDiseaseInfectious | 1074
HumanGeneticDiseases | 11468
RareSubset | 15682
Numbers look ok. 
2 changes requested: 
- Can we please update the name of this query to indicate that it is "**human**-rare disease"?
- Can we add also number of cancer terms? (it looks like it is missing in the original spreadsheet, but it is in the statistics that we often report (e.g. [here](https://mondo.monarchinitiative.org/))

3) non-human genetic diseases: src/sparql/reports/COUNT-non-human-genetic-diseases.sparql
- the count is : 1021    (@katiermullen, does this number make sense to you?)

Changes requested: 
In the report. the name is "?countHumanGeneticDiseases" instead of "count**NON**HumanGeneticDiseases"

4. sh run.sh make report-reason-query-COUNT-non-human_diseases_infectious
?countNonHumanDiseaseInfectious = 87. (@katiermullen does this number make sense to you?

sabrinatoro commented:
Someone should confirm the changes in the mondo.Makefile are ok. @matentzn, could you please check and/or confirm it is ok? Thank you

twhetzel commented:
@sabrinatoro I pushed new changes based on your comments above that include these updates:
Issue 1: Human disease + non-human disease are not equal to all disease excluding susceptibility.
Update: The count of "human disease" now excludes any classes that are also a subclass of susceptibility

Issue 2: Update the name of this query to indicate that it is "human-rare disease"
I updated the name of the query to indicate it is human rare disease. 
I also updated the query to count subclasses of MONDO:0700096 'human disease' vs. MONDO:000001 disease that are in the rare subset. 

There is now a query to get the number of cancer terms

Changed variable name for COUNT-non-human-genetic-diseases.sparql to ?countNonHumanGeneticDiseases

TODO: Counts for non-human-genetic-diseases and non-human_diseases_infectious need to be reviewed by @katiermullen

katiermullen commented:
I can confirm that the counts for non-human-genetic-diseases (1021) and non-human_diseases_infectious (87) correspond to the number of terms in the 'hereditary disease, non-human animal' and 'infectious disease, non-human animal' branches, respectively.

sabrinatoro reviewed (changes requested):
I ran both 
`sh run.sh make create-general-mondo-stats-all` and `sh run.sh make create-synonym-mondo-stats-all`. 
Both got the following error: 
reports/mondo_stats/tmp/COUNT-all_disease_excluding_susceptibility.tsv] Error 1

I will run the other sparql individually and test if their work

sabrinatoro reviewed (commented):
report | count
-- | --
?countAllDiseaseExcludingSusceptibility | 25379
?countHumanDiseaseExcludingSusceptibility | 22419
?countNonHumanAnimalDisease | 2959

Sum of human and non-human diseases = 25378
There is 1 difference. I wonder if this is counting the term "disease"

The counts and titles in the reports are ok.

twhetzel commented:
Thanks for reviewing @sabrinatoro. The count queries use `?class rdfs:subClassOf* obo:MONDO_0000001` so with the `*` the query is counting the class itself and all of it's subclasses.

For the error running the `make` goals, is this, `reports/mondo_stats/tmp/COUNT-all_disease_excluding_susceptibility.tsv] Error 1` the only error message that is displayed or were there any other lines above this?

twhetzel commented on src/sparql/reports/COUNT-all_disease_excluding_susceptibility.sparql:
```diff
@@ -0,0 +1,23 @@
+      # MONDO:0042489 'disease susceptibility'
+      ?class rdfs:subClassOf* obo:MONDO_0042489 .
+    }
+  }
```
Nico - from Tech call, this `MINUS` statement should be removed from ALL queries.

twhetzel commented:
@sabrinatoro this is ready to review with the updates from the Tech call discussion. The initial post has all `make` commands to run.

sabrinatoro commented:
@twhetzel I am still having an error: when running `sh run.sh make create-general-mondo-stats-all`

Here is the full report: 

v1.5.3: Pulling from obolibrary/odkfull
Digest: sha256:2ab1cc4e0d2fa6289051f245e889aeeab9ceffaa2bc01599aa95a85d759a5028
Status: Image is up to date for obolibrary/odkfull:v1.5.3
docker.io/obolibrary/odkfull:v1.5.3
Running ODK with 8 GB of memory, using odkfull:v1.5.3.
Makefile:723: warning: overriding recipe for target 'mirror-hgnc'
Makefile:607: warning: ignoring old recipe for target 'mirror-hgnc'
mondo.Makefile:1559: warning: overriding recipe for target 'help'
Makefile:1462: warning: ignoring old recipe for target 'help'
perl -ne 'print unless (m@^xref: (Orphanet|OMIM|OMIMPS|DOID|EFO|NCIT|SCTID|MESH|UMLS|ICD10CM|icd11.foundation):@ && !(m@(obsoleteEquivalent|equivalentObsolete|equivalentTo|relatedTo|mondoIsNarrowerThanSource|directSiblingOf|mondoIsBroaderThanSource)@i))' mondo-edit.obo | grep -v '^property_value: excluded_subClassOf' | egrep -v '^synonym: .*EXCLUDE' | egrep -v 'relationship: disease_has_basis_in_dysfunction_of (hgnc|HGNC|NCBIGene):' > filtered.obo.tmp && mv filtered.obo.tmp filtered.obo
perl -ne 'print if !(/^xref: (Orphanet|OMIM|OMIMPS|DOID):/ && !(/(obsoleteEquivalent|equivalentObsolete|equivalentTo|obsoleteEquivalentObsolete)/i))' filtered.obo  > filtered.obo.tmp && mv filtered.obo.tmp filtered.obo
if [ false = false ]; then robot --catalog catalog-v001.xml query -i filtered.obo --use-graphs false \
	--update ../sparql/update/delete-axiom-annotations.ru \
	--update ../sparql/update/delete-axiom-annotations-by-prefix.ru \
	--update ../sparql/update/inject-cross-species-analog.ru \
	remove --term-file config/remove-annotations-before-release.txt \
	convert -f obo --check false -o filtered.obo.obo && mv filtered.obo.obo filtered.obo; fi
../utils/mk-skos.pl filtered.obo > skos.ttl
owltools --use-catalog filtered.obo  skos.ttl --merge-support-ontologies -o filtered.owl
adding support ont Ontology(OntologyID(Anonymous-11)) [Axioms: 108102 Logical Axioms: 0] First 20 axioms: {AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0011991> <http://identifiers.org/medgen/330838>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0015731> <http://identifiers.org/snomedct/253771003>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0014585> <http://linkedlifedata.com/resource/umls/id/C4225370>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0012407> <http://identifiers.org/mesh/C566449>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0011051> <http://identifiers.org/medgen/330467>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0014068> <https://omim.org/entry/615163>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0008080> <http://linkedlifedata.com/resource/umls/id/C2931480>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0013821> <http://purl.obolibrary.org/obo/DOID_0070046>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0017603> <http://linkedlifedata.com/resource/umls/id/C1332078>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_8000010> <http://www.orpha.net/ORDO/Orphanet_80>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0013051> <https://omim.org/entry/612940>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0010210> <http://purl.obolibrary.org/obo/DOID_0110843>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0015329> <http://identifiers.org/snomedct/205808005>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0019630> <http://linkedlifedata.com/resource/umls/id/C1303012>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0010852> <http://identifiers.org/medgen/333071>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0014119> <http://purl.obolibrary.org/obo/DOID_0081099>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0958242> <http://identifiers.org/medgen/1853928>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0014557> <http://www.orpha.net/ORDO/Orphanet_459033>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0019695> <http://www.orpha.net/ORDO/Orphanet_93436>) AnnotationAssertion(<http://www.w3.org/2004/02/skos/core#exactMatch> <http://purl.obolibrary.org/obo/MONDO_0019161> <http://identifiers.org/snomedct/43941006>) }
robot --catalog catalog-v001.xml reason -T true -x true -X true -i filtered.owl -r ELK relax reduce -p false -r ELK \
	remove --term MONDO:0700097 --term OGMS:0000031 annotate -V http://purl.obolibrary.org/obo/mondo/releases/`date +%Y-%m-%d`/reasoned.owl.owl -o reasoned.owl
robot --catalog catalog-v001.xml merge -i reasoned.owl annotate -V http://purl.obolibrary.org/obo/mondo/releases/`date +%Y-%m-%d`/mondo.owl -o mondo.owl
mkdir -p reports/mondo_stats/mondo-rare-stats
Running query ../sparql/reports/COUNT-all_diseases.sparql 
robot --catalog catalog-v001.xml query -i mondo.owl --use-graphs true  -f tsv --query ../sparql/reports/COUNT-all_diseases.sparql reports/mondo_stats/tmp/COUNT-all_diseases.tsv
reports/mondo_stats/tmp/COUNT-all_diseases.tsv (No such file or directory)
Use the -vvv option to show the stack trace.
Use the --help option to see usage information.
make: *** [mondo.Makefile:375: reports/mondo_stats/tmp/COUNT-all_diseases.tsv] Error 1

twhetzel commented:
@sabrinatoro thanks, I will look into that error more. 

Attached are the GitHub stats files, issue status and usernames, for the time period of 2025-02-04 to 2025-03-04 (the last two tagged mondo releases). 
[gh_issue_status.csv](https://github.com/user-attachments/files/19542352/gh_issue_status.csv)
[gh_issue_usernames.csv](https://github.com/user-attachments/files/19542353/gh_issue_usernames.csv)

sabrinatoro reviewed (approved):
The statistics look good to me. 

The synonym reports was updated to "all diseases" (from "human disease" in a previous commit). I don't know whether this is what we want, ie have the synonym information all diseases versus separated for human and non-human. I will ask the team. 
However, this is not sufficient to hold this PR from being merge. We can update in a separate PR if needed. 

I am approving.

twhetzel commented:
Ok, thanks for checking the data. I am/have a fix for the error you ran into and also making a change so the data is generated based on the date of the most recent tagged release or a tag release date the user adds into the commandline argument.

sabrinatoro commented:
@twhetzel I thought about it. If it is not difficult, let's have the statistics for the synonyms for all diseases, human diseases, and non-human diseases. This way, we have it. 
For simplicity of review, this can be done in a separate PR (after this one is merged). 
Thank you! This is great to have all of these!

twhetzel commented:
@matentzn can you review the make goals (code and testing) and the queries in this PR?

matentzn commented on .github/workflows/github-issue-stats.yml:
```diff
@@ -0,0 +1,53 @@
+
+on:
+  workflow_dispatch:
+    inputs:
```
Never seen this in my life - how can this be used? Thats cool! When I trigger the workflow on GitHub, is it basically asking me for these inputs?

twhetzel replied:
yes, that is how it should work. this final test of the workflow can only be done once the PR is on `master` since it's a new workflow!

matentzn commented on src/ontology/mondo.Makefile:
```diff
@@ -265,6 +265,169 @@ create-mondo-stats:
+    $(SPARQLDIR)/reports/COUNT-non-human_diseases_infectious.sparql \
+    $(SPARQLDIR)/reports/COUNT-non-human_diseases_cancer.sparql
+
+RARE_SATISTICS_QUERIES = \
```
Typo SATISTICS

matentzn reviewed (approved):
I didn't test it but it looks good. Minor typo if you care, I and fine with it.

twhetzel commented:
@matentzn did you review all the sparql queries too? Also, since `mondo.owl` is being downloaded from a release tag, does it still need to be processed by other `make` goals?
